### PR TITLE
UI: Onboard template to domain

### PR DESCRIPTION
**Pull request by OferLevi85**
Signed-off-by: Ofer Levi <ofer.levi@verizonmedia.com>

### CODE REVIEW

File: servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java [hunk 1]
```diff
@@ -3074,9 +3074,13 @@ public DomainTemplateDetailsList getServerTemplateDetailsList(ResourceContext ct
         validateRequest(ctx.request(), caller);
 
         List<TemplateMetaData> serverTemplateMetadataList = serverSolutionTemplates.getTemplates()
-                .values()
+                .entrySet()
```
Comment (OferLevi85) [inline servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java:3077]: Added template names to the response

File: ui/src/components/template/TemplateList.js [hunk 2]
```diff
@@ -46,25 +46,88 @@ const TableHeadStyled = styled.th`
...
+            errorMessage: errorMessage,
+        });
+        // this is to close the failure alert
+        setTimeout(
+            () =>
+                this.setState({
+                    showError: false,
+                    errorMessage: '',
+                }),
+            MODAL_TIME_OUT
```
Comment (chandrasekhar1996) [inline ui/src/components/template/TemplateList.js:103]: we shouldn't use timed close for errors. It should persist on the screen until user actively dismisses it.

File: ui/src/components/template/TemplateList.js [hunk 3]
```diff
@@ -157,6 +253,28 @@ export default class TemplateList extends React.Component {
...
                     </thead>
                     <tbody>{rows}</tbody>
+                    <TitleDiv>Server Templates</TitleDiv>
+                    <thead>
+                        <tr>
+                            <TableHeadStyled align={left}>
+                                TEMPLATE
+                            </TableHeadStyled>
+                            <TableHeadStyled align={left}>INFO</TableHeadStyled>
+                            <TableHeadStyled align={center}>
```
Comment (chandrasekhar1996) [inline ui/src/components/template/TemplateList.js:263]: if the server template list only contains the templates which haven't be applied to the domain then do we require CurrentVersion column?

Comment (OferLevi85) [reply ui/src/components/template/TemplateList.js:263]: It's for symmetry with the "existing templates" table above it.

Comment (chandrasekhar1996) [reply ui/src/components/template/TemplateList.js:263]: ohh in that case it would be better to show 0 or N/A as value instead of leaving it empty

Comment (OferLevi85) [reply ui/src/components/template/TemplateList.js:263]: Will place N/A

File: ui/src/components/template/TemplateList.js (hunk 2 above)
Comment (OferLevi85) [reply ui/src/components/template/TemplateList.js]: Fixed

File: ui/src/components/template/TemplateList.js (hunk 3 above)
Comment (OferLevi85) [reply ui/src/components/template/TemplateList.js:257]: Fixed

**Review by chandrasekhar1996 — APPROVED** [review]

**Pull request merged**